app/decorators/service_provider_session_decorator.rb

```diff
@@ -48,4 +52,18 @@ def sp_return_url
   private
 
   attr_reader :sp, :view_context
+
+  def sp_logo_file_exist?
```

Comment:
Yea, a broken image is fine. That way there are three distinct cases: 1) Logo path not provided, use generic. 2) Logo path is provided, use it. 3) Logo path is wrong, show broken image.